File: 11-4/backend/schemas/organization.py
from datetime import datetime, time
from typing import Optional
from pydantic import BaseModel, Field


class OrganizationCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    type: str = Field(..., min_length=1, max_length=20)


class OrganizationResponse(BaseModel):
    id: int
    name: str
    type: str
    adminId: int
    createdAt: datetime
    memberCount: Optional[int] = 0

    class Config:
        from_attributes = True


class OrganizationMemberAdd(BaseModel):
    userId: str = Field(..., min_length=1)


class OrganizationMemberResponse(BaseModel):
    id: int
    organizationId: int
    userId: int
    userName: str
    userUserId: str
    role: str
    joinedAt: datetime

    class Config:
        from_attributes = True


class AttendanceCheckRequest(BaseModel):
    organizationId: int
    image: str = Field(..., description="Base64 encoded image")


class AttendanceStatsResponse(BaseModel):
    totalMembers: int
    todayCount: int
    participationRate: float
    records: list


class OrganizationDetailResponse(BaseModel):
    id: int
    name: str
    type: str
    adminId: int
    createdAt: datetime
    members: list[OrganizationMemberResponse]

    class Config:
        from_attributes = True

File: 15-1/frontend/src/types/organization.ts
export interface OrganizationCreate {
  name: string
  type: string
}

export interface OrganizationResponse {
  id: number
  name: string
  type: string
  adminId: number
  createdAt: string
  memberCount?: number
}

export interface OrganizationMemberAdd {
  userId: string
}

export interface OrganizationMemberResponse {
  id: number
  organizationId: number
  userId: number
  userName: string
  userUserId: string
  role: string
  joinedAt: string
}

export interface OrganizationDetailResponse {
  id: number
  name: string
  type: string
  adminId: number
  createdAt: string
  members: OrganizationMemberResponse[]
}

export interface AttendanceCheckRequest {
  organizationId: number
  image: string
}

export interface AttendanceRecord {
  id: number
  userId: number
  userName: string
  checkInTime: string
  status: string
}

export interface AttendanceStatsResponse {
  totalMembers: number
  todayCount: number
  participationRate: number
  records: AttendanceRecord[]
}